api/src/user_enable.ts

```diff
@@ -0,0 +1,123 @@
+import { Ctx } from "./lib/ctx";
+import * as Result from "./result";
+import { ServiceUser } from "./service/domain/organization/service_user";
+import * as UserEnable from "./service/domain/organization/user_enable";
+
+interface RequestBodyV1 {
+  apiVersion: "1.0";
+  data: {
+    userId: string;
+  };
+}
+
+const requestBodyV1Schema = Joi.object({
+  apiVersion: Joi.valid("1.0").required(),
+  data: Joi.object({
+    userId: Joi.string().required(),
+  }).required(),
+});
+
+type RequestBody = RequestBodyV1;
+const requestBodySchema = Joi.alternatives([requestBodyV1Schema]);
+
+function validateRequestBody(body: any): Result.Type<RequestBody> {
+  const { error, value } = Joi.validate(body, requestBodySchema);
+  return !error ? value : error;
+}
+
+function mkSwaggerSchema(server: FastifyInstance) {
+  return {
+    beforeHandler: [(server as any).authenticate],
+    schema: {
+      description: "Enable an user account",
+      tags: ["user"],
+      summary: "Enable an user account",
+      security: [
+        {
+          bearerToken: [],
+        },
+      ],
+      body: {
+        type: "object",
+        required: ["apiVersion", "data"],
+        properties: {
+          apiVersion: { type: "string", example: "1.0" },
+          data: {
+            type: "object",
+            required: ["userId"],
+            properties: {
+              userId: { type: "string", example: "aSmith" },
+            },
+          },
+        },
+      },
+      response: {
+        200: {
+          description: "successful response",
+          type: "object",
+          properties: {
+            apiVersion: { type: "string", example: "1.0" },
+            data: {
+              type: "object",
+            },
+          },
+        },
+        401: NotAuthenticated.schema,
+      },
+    },
+  };
+}
+
+interface Service {
+  enableUser(
+    ctx: Ctx,
+    issuer: ServiceUser,
+    issuerOrganization: string,
+    requestData: UserEnable.RequestData,
+  ): Promise<void>;
+}
+
+export function addHttpHandler(server: FastifyInstance, urlPrefix: string, service: Service) {
+  server.post(`${urlPrefix}/global.enableUser`, mkSwaggerSchema(server), (request, reply) => {
+    const ctx: Ctx = { requestId: request.id, source: "http" };
+
+    const issuer: ServiceUser = {
+      id: (request as AuthenticatedRequest).user.userId,
+      groups: (request as AuthenticatedRequest).user.groups,
+    };
+    const issuerOrganization: string = (request as AuthenticatedRequest).user.organization;
+    const bodyResult = validateRequestBody(request.body);
+
+    if (Result.isErr(bodyResult)) {
+      const { code, body } = toHttpError(new VError(bodyResult, "failed to enable an user"));
+      reply.status(code).send(body);
+      return;
+    }
+
+    const revokee = {
+      userId: bodyResult.data.userId,
+    };
+
+    service
+      .enableUser(ctx, issuer, issuerOrganization, revokee)
+      .then(() => {
```

Comment:
handle returned result here instead of promise

Reply:
done